base/configurable.py

```diff
@@ -52,9 +42,9 @@ class ConfigurableType(Enum):
 class Configurable:
     """Defines a class which the config of its instances can be managed by Config class"""
 
-    def __init__(self, bot):
+    def __init__(self, bot: BaseBot):
         self.iodirs = {}
-        self.bot = bot
+        self.bot: BaseBot = bot
```

Comment:
Maybe check which Plugins still have a self.bot = bot (and self.can_reload = true)